tests/resources/test_framework_definition_processing.py

```diff
@@ -0,0 +1,151 @@
+    frameworks = Namespace(
+        gama=Namespace(name="gama", version="latest", extends=None),
+    )
+    find_all_parents(frameworks.gama, frameworks)
+
+
+@pytest.mark.parametrize("framework", ["gama", "h2o_automl"])
+def test_find_all_parents_returns_parent_of_framework_with_single_parent(framework):
+    frameworks = Namespace(
+        gama=Namespace(name="gama", version="latest", description="flexible automl"),
+        gama_old=Namespace(name="gama_20.1.0", version="20.1.0", extends="gama"),
+        h2o_automl=Namespace(name="h2o", version="latest"),
+        h2o_automl_old=Namespace(name="h2o_1.2", version="1.2", extends="h2o_automl"),
+    )
+    parents = find_all_parents(frameworks[f"{framework}_old"], frameworks)
+    assert parents == [frameworks[framework]]
+
+
+@pytest.mark.parametrize("framework", ["gama", "h2o_automl"])
+def test_find_all_parents_returns_frameworks_closest_first_if_two_parents(framework):
+    frameworks = Namespace(
+        gama=Namespace(name="gama", version="latest", description="flexible automl"),
+        gama_old=Namespace(name="gama_20.1.0", version="20.1.0", extends="gama"),
+        gama_older=Namespace(name="gama_20.0.0", version="20.0.0", extends="gama_old"),
+        h2o_automl=Namespace(name="h2o", version="latest"),
+        h2o_automl_old=Namespace(name="h2o_1.2", version="1.2", extends="h2o_automl"),
+        h2o_automl_older=Namespace(name="h2o_1.1", version="1.1", extends="h2o_automl_old"),
+    )
+    parents = find_all_parents(frameworks[f"{framework}_older"], frameworks)
+    assert parents == [frameworks[f"{framework}_old"], frameworks[framework]]
+
+
+def test_update_frameworks_with_parent_definitions_add_missing_field_from_parent():
+    frameworks = Namespace(
+        gama=Namespace(name="gama", version="latest", description="flexible automl"),
+        gama_old=Namespace(name="gama_20.1.0", version="20.1.0", extends="gama"),
+    )
+    update_frameworks_with_parent_definitions(frameworks)
+    assert frameworks.gama_old.description == "flexible automl"
+
+
+def test_update_frameworks_with_parent_definitions_does_not_overwrite_child_yaml():
+    frameworks = Namespace(
+        gama=Namespace(name="gama", version="latest", description="flexible automl"),
+        gama_old=Namespace(name="gama_20.1", version="20.1", description="old gama", extends="gama"),
+    )
+    update_frameworks_with_parent_definitions(frameworks)
+    assert frameworks.gama_old.description == "old gama"
+
+
+def test_update_frameworks_with_parent_definitions_parent_overwrites_grandparent_yaml():
+    frameworks = Namespace(
+        gama=Namespace(name="gama", version="latest", description="flexible automl"),
+        gama_old=Namespace(name="gama_2", version="1", description="automl", extends="gama"),
+        gama_oldest=Namespace(name="gama_1", version="2", extends="gama_old"),
+    )
+    update_frameworks_with_parent_definitions(frameworks)
+    assert frameworks.gama_oldest.description == "automl"
```

Comment:
Also, would be nice to have one test checking that the `params` field is not merged, but fully inherited or fully overridden, just in case we mess up this in the future.
```
    frameworks = Namespace(
        gama=Namespace(name="gama", version="latest", params=dict(foo='bar')),
        gama_old=Namespace(name="gama_2", version="1", extends="gama"),
    )
     update_frameworks_with_parent_definitions(frameworks) 
     assert frameworks.gama_old.params ==  frameworks.gama.params
```
and 
```
    frameworks = Namespace(
        gama=Namespace(name="gama", version="latest", params=dict(foo='bar')),
        gama_old=Namespace(name="gama_2", version="1", extends="gama", params=dict(bar='baz')),
    )
     update_frameworks_with_parent_definitions(frameworks) 
     assert frameworks.gama_old.params ==  dict(bar='baz')
```

Reply:
Parametrized the tests to work with any field, added `params`  and `setup_args`. But in principle it is just testing the same thing, as the way the adding of defaults is done is slightly different now. Previously the inheritance took place after parents had their defaults set, while now it takes place before.